### PR TITLE
Media More Info: Convert to Lit Element

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

Convert to Lit 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

### CODE REVIEW

File: src/dialogs/more-info/controls/more-info-media_player.ts [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+                      @click=${this._handleClick}
+                    ></ha-icon-button>
+                  `
+                : ""}
+            </div>
+          `
+        : ""}
+      ${stateObj.state === "off" ||
+      !supportsFeature(stateObj, SUPPORT_SELECT_SOURCE) ||
+      !stateObj.attributes.source_list
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-media_player.ts:119]: Can we turn this around?
```suggestion
      ${stateObj.state !== "off" &&
      supportsFeature(stateObj, SUPPORT_SELECT_SOURCE) &&
      stateObj.attributes.source_list
```

Comment (bramkragten) [reply src/dialogs/more-info/controls/more-info-media_player.ts:119]: And should we check if `source_list` has a length?

File: src/dialogs/more-info/controls/more-info-media_player.ts [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+                <paper-listbox
+                  slot="dropdown-content"
+                  attr-for-selected="item-name"
+                  .selected=${stateObj.attributes.source!}
+                  @iron-select=${this._handleSourceChanged}
+                >
+                  ${stateObj.attributes.source_list!.map(
+                    (source) =>
+                      html`
+                        <paper-item itemName=${source}>${source}</paper-item>
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-media_player.ts:136]:
```suggestion
                        <paper-item .itemName=${source}>${source}</paper-item>
```

File: src/dialogs/more-info/controls/more-info-media_player.ts [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+    const services = this.hass.services.tts;
+    const serviceKeys = Object.keys(services).sort();
+    let service: string | undefined;
+
+    for (let i = 0; i < serviceKeys.length; i++) {
+      if (serviceKeys[i].indexOf("_say") !== -1) {
+        service = serviceKeys[i];
+        break;
+      }
+    }
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-media_player.ts:364]: Ugh, this is so ugly, we should look into it sometime...

File: src/dialogs/more-info/controls/more-info-media_player.ts [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+
+    if (!service || !this._ttsInput) {
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-media_player.ts:366]: Make a const for ttsInput:
`const ttsInput = this._ttsInput;`
You are querying the dom 3 times now (new LitElement will have a cache option :tada:)

File: src/dialogs/more-info/controls/more-info-media_player.ts [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+  private _sendTTS() {
+    const ttsInput = this._ttsInput;
+    if (!ttsInput) {
+      return;
+    }
+
+    const services = this.hass.services.tts;
+    const serviceKeys = Object.keys(services).sort();
+
+    const service = serviceKeys.find((key) => key.indexOf("_say") !== -1);
```
Comment (zsarnett) [inline src/dialogs/more-info/controls/more-info-media_player.ts:363]: @bramkragten This should be a bit better than the for loop I think. Still maybe not the best way

Comment (bramkragten) [reply src/dialogs/more-info/controls/more-info-media_player.ts:363]: the fact that we just pick the first one is wrong, to begin with... But yeah, this code is a lot better

**Review by bramkragten — APPROVED** [review]

File: src/dialogs/more-info/controls/more-info-media_player.ts (hunk 5 above)
Comment (zsarnett) [reply src/dialogs/more-info/controls/more-info-media_player.ts:363]: Yea... that's what I was thinking. We also sort them... but from the data, we are given from the service there doesn't seem to be a way to find the preferred one

**Pull request merged**